lib/schemas.js

```diff
@@ -30,6 +30,7 @@ const HANDLER_OPTIONS = joi
     parentField: FIELD,
     pickFields: [FIELD, joi.array().min(1).items(FIELD)],
     versionField: FIELD,
+    versionResolver: joi.func(),
```

Comment:
I'd like to see `versionField` and `versionResolver` as mutually exclusive fields, like:
`.oxor('versionField', 'versionResolver')`